internal/archive/archive.go

```diff
@@ -98,15 +95,15 @@ func (a *ubuntuArchive) selectPackage(pkg string) (control.Section, *ubuntuIndex
 		}
 	}
 	if selectedVersion == "" {
-		return nil, nil, fmt.Errorf("cannot find package %q in archive", pkg)
+		return nil, nil, ""
 	}
-	return selectedSection, selectedIndex, nil
+	return selectedSection, selectedIndex, selectedVersion
 }
 
 func (a *ubuntuArchive) Fetch(pkg string) (io.ReadCloser, error) {
-	section, index, err := a.selectPackage(pkg)
-	if err != nil {
-		return nil, err
+	section, index, _ := a.selectPackage(pkg)
+	if section == nil {
+		return nil, fmt.Errorf("cannot find package %q in archive", pkg)
```

Comment:
That means we not only implicitly know that section might be nil, but we can even tell the only possible reason why it might be nil, which is not great. Can we please preserve the proper error return from selectPackage, and also have Version returning an error?  We may also introduce a type NotFoundError so that the result may be compared against it with errors.Is, so we can disambiguate where it makes sense.